Gems/Multiplayer/Code/Include/Multiplayer/Components/SimplePlayerSpawnerComponentBus.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/EBus/EBus.h>
+#include <AzCore/Outcome/Outcome.h>
+#include <AzCore/std/string/string.h>
+
+namespace Multiplayer
+{
+    //! @class SimplePlayerSpawnerRequests
+    //! @brief The SimplePlayerSpawnerRequest event-bus exposes helper methods regarding network player spawners.
+    //! Although the Multiplayer System automatically spawns in players, it's common for game specific server logic to retrieve valid spawn locations when respawning a player.
+    class SimplePlayerSpawnerRequests : public AZ::EBusTraits
+    {
+    public:
+        static const AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Single;
+        static const AZ::EBusAddressPolicy AddressPolicy = AZ::EBusAddressPolicy::Single;
+        
+        //! Returns the location where the next joining player will be spawned, and then advances the spawn point index.
+        //! The index will wrap back to 0 in a round-robin fashion so that a valid spawn point is always available.
+        //! Method is only valid if called from the multiplayer host/authority; clients are not given information regarding the spawn point index.
+        //! @return AZ::Transform The location of the next spawn point
+        virtual AZ::Transform RoundRobinNextSpawnPoint() = 0;
```

Comment:
Suggestion: I'd still suggest "Advance" over "RoundRobin" but it depends on your goals for the API. 

My thinking is that the actual selection algorithm would ideally be data-driven, and not "baked" into the different API names. For example, you could imagine having a dropdown on the component that says "RoundRobin" / "Random" / "Farthest" / "Nearest". In that case, you'd want the API to use whatever was on the dropdown.

However, I could see an alternate argument where the dropdown could contain the _default_ and you want the API to override it. Though even in that case, I'd think the API would take in an enum of algorithms instead of having a different call per algorithm.

A third point to consider - even if you _don't_ want multiple algorithms, by exposing the choice of algorithm in the name of the API, it means that if you ever want to change which algorithm this component uses, the API name will need to change.

Reply:
Very true, now with this wording this API can be generic and used for other spawner class. I wonder if I should rename this to just _PlayerSpawnerBus.h_?

Reply:
To follow up, I actually just removed the "advance" option. Avoids added complexity, and a game can simply Get/Set the index to their liking.